src/coprocessor/dag/executor/topn.rs

```diff
@@ -95,25 +96,32 @@ impl TopNExecutor {
             return Ok(());
         }
 
-        let mut heap = TopNHeap::new(self.limit)?;
+        let mut heap = TopNHeap::new(self.limit, Arc::clone(&self.ctx.cfg))?;
         while let Some(row) = self.src.next()? {
             let cols = inflate_with_col_for_dag(
-                &self.ctx,
+                &mut self.ctx,
                 &row.data,
                 self.cols.as_ref(),
                 &self.related_cols_offset,
                 row.handle,
             )?;
-            let ob_values = self.order_by.eval(&self.ctx, &cols)?;
+            let ob_values = self.order_by.eval(&mut self.ctx, &cols)?;
             heap.try_add_row(
                 row.handle,
                 row.data,
                 ob_values,
                 Arc::clone(&self.order_by.items),
-                Arc::clone(&self.ctx),
             )?;
         }
-        self.iter = Some(heap.into_sorted_vec()?.into_iter());
+        let sort_rows = heap.into_sorted_vec()?;
+        let data: Vec<Row> = sort_rows
```

Comment:
Why?

Reply:
I would like to store `Row` instead of `SortRow` into `TopNExecutor`, and use `TopNHeap` and `SortRow` only in function `fetch_all`. So the `Arc<EvalConfig>` and `Arc<Vec<ColumnInfo>>` in `SortRow` can be released earlier and the `TopNexecutor` can be easier.